kernel/include/kernel_internal.h

```diff
@@ -51,6 +51,24 @@ extern void z_setup_new_thread(struct k_thread *new_thread,
 			      void *p1, void *p2, void *p3,
 			      int prio, u32_t options, const char *name);
 
+#if defined(CONFIG_FLOAT) && defined(CONFIG_FP_SHARING)
```

Comment:
Style-wise: we don't generally predicate the declaration of an API on a kconfig.  It's enough that the function is #if'd out at the compile level and you get a linkage failure.  But I don't think we're super-consistent about that.

Reply:
This is internal API, right?
So, correct, @andyross, we are not at all consistent here: at least in two occasions
- the user mode -related internal APIs,
- the custom busy wait
the function declarations lie inside the respective #ifdef guards.
Therefore, the k_float_disable() is added with the appropriate #ifdefs.

The question is whether all of them should be declared unconditionally. I would be ok with that - if you've got a strong opinion, let's do it for all functions in a follow-up patch